pandas/_libs/groupby_helper.pxi.in

```diff
@@ -19,6 +19,18 @@ ctypedef fused rank_t:
     object
 
 
+cdef inline bint _treat_as_na(rank_t val, bint is_datetimelike) nogil:
+    if rank_t is object:
+        # Should never be used, but we need to avoid the `val != val` below
+        #  or else cython will raise about gil acquisition.
```

Comment:
this is pretty odd, why dont you return an enum here (true, false, raise) and handle in code appropriately.

Reply:
the vicissitudes of cython fused types